### ISSUE TITLE
[New build - EXPERIMENTAL] 23w48a: Early Game Components

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#5363]
This update brings several components needed for the survival early game.

Here are the new blocks and features in a nutshell:
- Personal crafting.
- New jump/boost jetpack mode.
- More wood shapes and blocks. Wood retains original tree color.
- Crucible: can smelt ore into ingots
- Crafting bench.
- Wooden crate.
- Wind Turbine.
- Stone block, stone retains original rock color.
- Coal spawn like other ore. Coal be used to fuel the crucible

Personal Crafting
The player can now craft a few basic schematics from its inventory menu.
More survival pratical items will be added over time.
The crafting here is purely manual (so it is made to be a bit tedious)

Crafting Bench
The crafting bench is a stepping stone between the personal crafting and the full fledged assembler.
Like the personal crafting it is manual.

Crucible.
Like the furnace the crucible can smelt ore into ingots. 
Unlike the Arc Furnace it do not require access to a powered grid but will require coal or wood to process ingots.
It also can only process a few ore type compared to the furnace.
It is possible to queue up several ingots.

Wood and coal.
It is now possible to destroy tree to get wood. Wood can be used in various schematics. 
More wood shapes have been added.
The wood will retain the color of the original tree, so it is up to you to explore the galaxy to get wood of the color you fancy!

Stone blocks:
Stone block can be crafted from rock. Like the tree they will retains the original rock color.

Wind Turbine
The wind turbine is a power generator available early from the crafting bench. It requires wind to produce power.

Jetpack refactor:
Obviously being able to levitate infinitely is not doable for survival, so a new jetpack mode is introduced: the jump-pack.
Jump-pack will provide a boost to the player jump, but also a dash forward when pressing shift.
I am not 100% satisfied with the dash, the keymapping will likely change.
Anti-g and 6-dof are still accessible in 0 or low g environment.

Bugfixes:
#5351 Various issues with solar panels

**Comment by Dwarf-LordPangolin** [comment]
Looks amazing! I quite like the way the jump-pack performs in gravity; the dash seems like it will be quite useful in evading attacks.

One potential issue I am seeing currently is that moving in zero-g now causes very significant view sway depending on the direction one moves + the angle up or down one is looking; that might cause motion sickness in some players. The sway should probably be removed while in zero-g; I think players trying to place blocks in space will get frustrated having their view moved while building.

Can't wait to look at all these survival features! 🚀

**Comment by ProPeach** [comment]
Also, are you not planning on adding the stone/wood variant colours to the creative codex? It's be really handy to have them there, although I'm planning on putting a stone/wood pallet on the workshop for convenience in the short term

For survival I really like the thought of exploring new worlds to get new material types, but creative has kinda got to have all the blocks at your disposal surely. On that note being able to spawn the material items/gas bottles etc in creative would also be great, and make things like testing out features and isolating bugs a lot easier

Side note, I'd LOVE to see some really dark stone colours! A deep black basalt like the terrain in Iceland, where the contrast between rich greens and dark black is really lovely. Could be an awesome biome for a planet!
![image](https://github.com/tsunamayo/Starship-EVO/assets/27857711/0471ed5c-4069-46d7-8ca6-85fe2f2573b7)

**Comment by TIKIRobo** [comment]
I agree with Dwarf, I think that the camera movement is a bit too much in 0g, a good compromise could be only change the FOV when "sprinting" in 0g, another option could be to disable the fov change when at more than 1x speed as at higher speed cheats it becomes quite disorientating

**Comment by BigBadKangaroo** [comment]
Stone and wood blocks are a great addition! I love them! But all Hull-Block types should have a stone/wood variant and we should be able to paint them.

I really love the new crafting bench. Now we can build tables that have a function and fit in so many ships.
![20231201171957_1](https://github.com/tsunamayo/Starship-EVO/assets/81495956/0ce6dd99-5f47-4cba-a0d1-408baefc5929)

Regarding the new jetpack mode: I like the addition of it, works out really nice. But the FOV effect when sprinting is really a bit too much imo.

**Comment by tsunamayo** [comment]
> Also, are you not planning on adding the stone/wood variant colours to the creative codex? It's be really handy to have them there, although I'm planning on putting a stone/wood pallet on the workshop for convenience in the short term
> 
> For survival I really like the thought of exploring new worlds to get new material types, but creative has kinda got to have all the blocks at your disposal surely. On that note being able to spawn the material items/gas bottles etc in creative would also be great, and make things like testing out features and isolating bugs a lot easier
> 
> Side note, I'd LOVE to see some really dark stone colours! A deep black basalt like the terrain in Iceland, where the contrast between rich greens and dark black is really lovely. Could be an awesome biome for a planet! Iceland was used as a filming location for the opening of Rogue One, so it certainly fits the bill for a non-Earth planet too ; ) ![image](https://private-user-images.githubusercontent.com/27857711/287291457-0471ed5c-4069-46d7-8ca6-85fe2f2573b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0NzYzMzMsIm5iZiI6MTcwMTQ3NjAzMywicGF0aCI6Ii8yNzg1NzcxMS8yODcyOTE0NTctMDQ3MWVkNWMtNDA2OS00NmQ3LThjYTYtODVmZTJmMjU3M2I3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAyVDAwMTM1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0Y2ViNzYyYTIyNTNhMDliODJjYjhjZmI5YzRmMjJkNGMwM2Q0OWVjNTNlYjU0NmY2ZmE3MGIxMjA2MTBiYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mEd_tyShzEZGcoYIeIpRMTeBa9w9VHdIIiMKnbztYDs)

Well stone / wood are simply using the paint to propagate their original color. So I could make them paintable in the creative mode I guess. But if I made them paintable in survival you loose the original color forever once you repaint it.
There should be some darker stone somewhere, once I saw a planet like that.

**Comment by tsunamayo** [comment]
So I will check the effect on the other jetpack mode, I will keep you posted. @BigBadKangaroo I do not want to add variant to every single block, it does not make sense practically it would clutter the codex, also you should not really need all those shape in wood or stone. But I am willing to add some more legit shapes, like a wood cylinder, or even a nice looking stone column of some kind.

**Comment by ProPeach** [comment]
> There should be some darker stone somewhere, once I saw a planet like that

I'll have to keep searching then! I guess that's part of why it'd be cool to see all the colours in the codex, I don't know the extent of colours that are out there lol

**Comment by tsunamayo** [comment]
@ProPeach Also you should jump much higher, I was at like 10m or so after a triple jump. You need to keep pressing the key though.

**Comment by tsunamayo** [comment]
> > There should be some darker stone somewhere, once I saw a planet like that
> 
> I'll have to keep searching then! I guess that's part of why it'd be cool to see all the colours in the codex, I don't know the extent of colours that are out there lol

Compared to other material stone have a more subtle hue shift but much larger brightness shift, so dark/light stone is possible.

**Comment by tsunamayo** [comment]
![SNAG-0809](https://github.com/tsunamayo/Starship-EVO/assets/6604318/92c86621-4864-4dcc-bf37-ee9b801a956f)
This one was one of the most ridiculous palette, but stone was darker but a bit purple too. I had one other in mind, I dont think I kept the screenshot though.

**Comment by Crimson-Artist** [comment]
@tsunamayo Just out of curiosity. Is Silicon ore still in the game? I have not encountered it since the planet update. it used to be that you used the normal rocks that spawned to make silicon but now since we officially have stone based blocks I assume that they will get their own ore vein.

**Comment by Gallt** [comment]
@tsunamayo Have you considered having buildable block shapes in radial (like Rust, see pic below) so you can have multiple materials/textures, with all blocks, without over-cluttering the codex? Things like truly decorative blocks (like vents, bars, ect) can stay their own thing. So you can slot Wood and Iron on 2 separate hotkeys, hold middle mouse button (or some hotkey), radial pops up, and you have the full bevvy of blocks. Probably would make adding full new sets easier?

I would like to see stone with higher HP though. Right now it has the same HP as wood (20 for 1m block) for 100x weight. Would be nice to make bunkers early game. Or maybe a material that takes 1 stone 1 iron to make and have it be like reinforced concrete to make stronger planetary buildings? The weight makes it difficult to justify for ships so i dont think it would unbalance anything.



![image](https://github.com/tsunamayo/Starship-EVO/assets/94942440/5447bc4e-c96c-4fee-8aa8-3d1acbe2078e)

**Comment by tsunamayo** [comment]
@BigBadKangaroo Hi, I fixed the shapes for next build.

**Comment by tsunamayo** [comment]
@Dwarf-LordPangolin Gotcha, tell me how it feel in 23w49a it is a different approach. It felt alright for me but I am willing to hear other opinion.
